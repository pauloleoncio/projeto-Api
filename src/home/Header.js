import React from 'react';
import {Link} from 'react-router-dom';
import logo from '../logo.svg';

export default function Header({titulo = 'Loja Virtual'}) {
  return (
    <header data-testid="header">
      <Link to={'/'}>
        <img src={logo} alt="logo do sites"/>
      </Link>
      <h1>{titulo}</h1>
    </header>
  );
}

