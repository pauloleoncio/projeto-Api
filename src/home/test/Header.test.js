import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Header';

test('Titulo do site foi exibido no Header', () => {
  const { getByText } = render(<Header />, {wrapper: MemoryRouter});
  const h1 = getByText(/Loja Virtual/i);
  expect(h1).toBeInTheDocument();
});

test('Ao informar um titulo, o header deve exibir este', () => {
  const {getByText } = render(<Header titulo="titulo de teste" />, {wrapper: MemoryRouter})
  const h1 = getByText(/titulo de teste/i);
  expect(h1).toBeInTheDocument();
});

test('Verifica a construção completa da tela', () => {
  const {getByTestId} = render(<Header />, {wrapper: MemoryRouter});
  const header = getByTestId('header');
  expect(header).toMatchSnapshot();
})