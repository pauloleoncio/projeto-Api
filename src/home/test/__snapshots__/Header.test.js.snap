// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verifica a construção completa da tela 1`] = `
<header
  data-testid="header"
>
  <a
    href="/"
  >
    <img
      alt="logo do site"
      src="logo.svg"
    />
  </a>
  <h1>
    Loja Virtual
  </h1>
</header>
`;
